### PR TITLE
Ports the Eggcellent Challenge from HS

### PR DESCRIPTION
**Pull request by Wallemations**
Beat the challenge, get the hat, save your dying friend!

Now with new **`NAMELESS HERO`** difficulty! _(Nameless hero difficulty does not gain you any additional benefits upon completing challenge.)_

### CODE REVIEW

**Comment by Wallemations** [comment]
what the hell happened that made this fail

File: maplestation_modules/code/modules/food_and_drinks/recipes/food/egg_recipes.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Egcellent Challenge
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/food_and_drinks/recipes/food/egg_recipes.dm:1]:
```suggestion
// Eggcellent Challenge
```

File: maplestation_modules/code/modules/food_and_drinks/food/eggs.dm [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		if(difficulty <= EGGS_TRUE_HERO && set_time >= 45 || !(difficulty <= EGGS_TRUE_HERO) && set_time >= 30)
+			var/below_time
+			switch(difficulty)
+				if(EGGS_TRUE_HERO || EGGS_NAMELESS_HERO)
+					below_time = 45
+				else
+					below_time = 30
+			. += span_tinynotice("To turn off Trial Run mode for the current difficulty, set timer below [below_time] minutes.")
+
+/obj/item/food/omelette/eggcellent_plate/proc/On_Consume(atom/eggs, mob/egg_eater, mob/egg_feeder)
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/food_and_drinks/food/eggs.dm:95]: WHY IS THIS WITH UPPERCASE WHAT ARE YOU DOIINNGGGGGG

**Review by CRITAWAKETS — CHANGES_REQUESTED** [review]
code is pretty good, though you'll wanna switch out the single-letter variable names

File: maplestation.dme [hunk 3]
```diff
@@ -5070,6 +5071,8 @@
 #include "maplestation_modules\code\modules\flavor_text\flavor_text.dm"
 #include "maplestation_modules\code\modules\flavor_text\pose_component.dm"
 #include "maplestation_modules\code\modules\food_and_drinks\drinks.dm"
+#include "maplestation_modules\code\modules\food_and_drinks\food\eggs.dm"
+#include "maplestation_modules\code\modules\food_and_drinks\recipes\food\egg_recipes.dm"
 #include "maplestation_modules\code\modules\food_and_drinks\recipes\drinks_recipes.dm"
```
Comment (CRITAWAKETS) [inline maplestation.dme:5076]:
```suggestion
#include "maplestation_modules\code\modules\food_and_drinks\recipes\drinks_recipes.dm"
#include "maplestation_modules\code\modules\food_and_drinks\recipes\food\egg_recipes.dm"
```

ah shit these need to be reversed too

**Review by CRITAWAKETS — CHANGES_REQUESTED** [review]

File: maplestation_modules/code/modules/food_and_drinks/food/eggs.dm [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		"spawn" = crown,
+	))
+
+/obj/item/food/omelette/eggcellent_plate/proc/spawn_bomb(mob/user)
+	var/obj/item/grenade/frag/P = new /obj/item/grenade/frag(user.loc)
+	P.active = TRUE
+	addtimer(CALLBACK(P, /obj/item/grenade/proc/detonate), rand(2,15))
+
+/obj/item/food/omelette/eggcellent_plate/proc/failed_eggs()
+	priority_announce("[current_challenger] has failed to finish [current_challenger.p_their()] quest in the given timeframe. Measures have been taken accordingly.", "Sacred Egg Enrichment Center")
```
[the remark this answers is not in the document]
Comment (Wallemations) [reply maplestation_modules/code/modules/food_and_drinks/food/eggs.dm:147]: I think I've got an idea for that

File: maplestation_modules/code/modules/food_and_drinks/food/eggs.dm [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+	addtimer(CALLBACK(failure_grenade, TYPE_PROC_REF(/obj/item/grenade, detonate), rand(0.2 SECONDS, 1.3 SECONDS)))
+
+/obj/item/food/omelette/eggcellent_plate/proc/failed_eggs()
+	var/mob/living/current_challenger = current_challenger_weak.resolve()
+	peer_pressure("[current_challenger] has failed to finish [current_challenger.p_their()] quest in the given timeframe. You feel no pity in what comes next.")
+	spawn_bomb(current_challenger)
+	qdel(src)
+
+/obj/item/food/omelette/eggcellent_plate/proc/peer_pressure(flavor_text)
+	loc.visible_message(flavor_text)
```
[the remark this answers is not in the document]
Comment (Wallemations) [reply maplestation_modules/code/modules/food_and_drinks/food/eggs.dm:154]: I didn't even think of that, I'll have to swap it out

**Pull request merged**